compiler/rustc_hir_typeck/src/fn_ctxt/mod.rs

```diff
@@ -144,7 +144,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 
-    pub(crate) fn dcx(&self) -> DiagCtxtHandle<'tcx> {
+    pub(crate) fn dcx(&self) -> DiagCtxtHandle<'a> {
```

Comment:
`'root_ctxt` (or `'rcx`) would be a better lifetime name than `'a` for `FnCtxt`.

Reply:
Preexisting and very verbose. I don't really think we should do it